great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -47,77 +41,61 @@ def __init__(
         )
 
         self._runtime_keys = runtime_keys
+        self._refresh_data_references_cache()
 
     def _refresh_data_references_cache(self):
-        """"""
-        # Map data_references to batch_definitions
-        data_reference: str = self._get_data_reference_list()[0]
-        mapped_batch_definition_list: List[
-            BatchDefinition
-        ] = self._map_data_reference_to_batch_definition_list(
-            data_reference=data_reference, data_asset_name=None
-        )
-        self._data_references_cache = {data_reference: mapped_batch_definition_list}
+        self._data_references_cache = {}
 
     def _get_data_reference_list(
         self, data_asset_name: Optional[str] = None
     ) -> List[str]:
-        """List objects in the underlying data store to create a list of data_references.
-
-        This method is used to refresh the cache.
         """
-        return [""]
+        List objects in the cache to create a list of data_references. If data_asset_name is passed in, method will
+        return all data_references for the named data_asset. If no data_asset_name is passed in, will return a list of
+        all data_references for all data_assets in the cache.
+        """
+        if data_asset_name:
+            return self._get_data_reference_list_from_cache_by_data_asset_name(
+                data_asset_name
+            )
+        else:
+            data_reference_list = [
+                self._get_data_reference_list_from_cache_by_data_asset_name(
+                    data_asset_name
+                )
+                for data_asset_name in self.get_available_data_asset_names()
+            ]
+            return data_reference_list
 
     def _get_data_reference_list_from_cache_by_data_asset_name(
         self, data_asset_name: str
     ) -> List[str]:
         """Fetch data_references corresponding to data_asset_name from the cache."""
-        batch_definition_list: List[
-            BatchDefinition
-        ] = self._get_batch_definition_list_from_batch_request(
-            batch_request=BatchRequestBase(
-                datasource_name=self.datasource_name,
-                data_connector_name=self.name,
-                data_asset_name=data_asset_name,
-            )
+        data_references_for_data_asset_name = self._data_references_cache.get(
+            data_asset_name
         )
-
-        if len(batch_definition_list) == 0:
-            return []
-        return [
-            self._map_batch_definition_to_data_reference(
-                batch_definition=batch_definition,
-            )
-            for batch_definition in batch_definition_list
-        ]
+        if data_references_for_data_asset_name is not None:
+            return list(data_references_for_data_asset_name.keys())
+        else:
+            return [""]
```

Comment:
@Shinnnyshinshin : Why not return `[]`?  Do we really need to return one element being an empty string?..

Reply:
I believe this was a carry-over from the previous behavior. Good point though. I cleaned this up in the most recent commit